### ISSUE TITLE
When baseURL is not a root URL, `absLangURL` adds language prefix when already present

### ISSUE THREAD

**Issue by shaform** [gohugoio/hugo#4995]
Previously, in #2444, @digitalcraftsman mentioned that for links added to content pages via their frontmatter in the menu, the language prefix is added twice.

This was fixed in latter commits.

However, when baseURL = `http://root-domain/directory` (i.e., not a root domain), then the problem reappears.

**Comment by bep** [comment]
We have had this discussion before, and the problem is that I don't know a way to distinguish a language code from ... a directory.

So, if the language code is `jp` and `baseURL = http://root-domain/jp` it gets ambiguous very fast.

**Comment by bep** [comment]
> how menu items are added by the frontmatter? For example, perhaps we could have a new field called .RelURL

Yes, something like that sounds good. But I would love a better name.

/cc @kaushalmodi

**Comment by bep** [comment]
@shaform I have thought about your problem, and I think I have a solution you can test:

As menu items come in two flavors:

1. Defined in Page (have language ready links)
2. In site config -- you only assume they have translations and use `relLangURL` or similar

`relLangURL` fails in 1) if you have a sub-folder in baseURL, but you can do:

```
{{ with $menuItem.Page }}
// use as is
{{ .RelPermalink }}
{{ else }}
{{ .URL | relLangURL }}
{{ end }}
```

**Comment by bep** [comment]
As an added note to the above: You could also do `.Site.GetPage` for the `config.toml`-defined menu items to check if they are present in the current language.

**Comment by shaform** [comment]
@bep Thank you very much! I've tested the code and it works. I think this solution is acceptable.

**Comment by javifm86** [comment]
I am experiencing this issue in version 0.64.

Code in my config.toml:

```
baseURL = "https://example.com/beta"

[languages.es.menu]
  [[languages.es.menu.main]]
    name = "Inicio"
    url = "/"
    weight = 1
  [[languages.es.menu.main]]
    name = "Blog"
    url = "/blog/"
    weight = 2

[languages.en.menu]
  [[languages.en.menu.main]]
    name = "Home"
    url = "/"
    weight = 1
  [[languages.en.menu.main]]
    name = "Blog"
    url = "/blog/"
    weight = 2
```

In my nav.html, as [suggested in docs](https://gohugo.io/content-management/multilingual/#menus):

```
<ul>
    {{- $currentPage := . -}}
    {{ range .Site.Menus.main -}}
    <li class="{{ if $currentPage.IsMenuCurrent "main" . }}active{{ end }}">
        <a href="{{ .URL | absLangURL }}">{{ .Name }}</a>
    </li>
    {{- end }}
</ul>
```

And when navigatng in English, links for navigation are wrong.

Link generated: http://localhost:1313/en/beta/blog/
Correct link: http://localhost:1313/beta/en/blog/

Thanks.

**Comment by jmooring** [comment]
This is resolved by using `pageRef` (introduced in v0.86.0) instead of `url` when defining menu entries in site configuration. Regardless of whether you define a menu entry in front matter or in your site configuration, each entry  has a page reference. The resulting URLs are language-relative. Do not pass them through `relLangURL` or `absLangURL`.

This is now a documentation issue, to be addressed shortly with https://github.com/gohugoio/hugoDocs/issues/1594.
